api/handler.go

```diff
@@ -301,11 +301,16 @@ func (h *apiHandler) queryUsage() http.HandlerFunc {
 		}
 
 		userId, ok := r.Context().Value(userIdContextKey).(string)
+
 		if !ok {
 			respondError(rw, http.StatusInternalServerError, errors.New("request not authenticated"))
 			return
 		}
 
+		if qs := r.URL.Query(); qs.Get("userId") != "" {
```

Comment:
Can you add a `// TODO` comment just to be sure? Please make sure not to deploy this to production anyway. Perhaps don't merge this just rebase `dev`?